dv-export/python/query-changes/command_handler.py

```diff
@@ -39,56 +49,94 @@ def __init__(self, dv_manager: DVManager, change_fetcher: ChangeFetcher, data_st
+        analytic_object_name = query[SOURCE][ANALYTIC_OBJECT]
+        logger.info(f"Analytic object to fetch changes: {analytic_object_name}")
 
         # Retrieve filter from query
         filter_name = self._get_filter_name(query)
-        filter_values = self._read_filter_values(export_uuid, analytic_object, filter_name) if filter_name else None
+        filter_values = self._read_filter_values(
+            export_uuid, analytic_object_name, filter_name) if filter_name else None
+        list_query_dto = self._create_list_query_dto(filter_values, query)
 
         # Retrieve analytic object changes
-        analytic_object_metadata, properties = self.change_fetcher.get_analytic_object_metadata(analytic_object)
+        analytic_object_dto, properties_dto = self.change_fetcher.get_analytic_object_metadata(analytic_object_name)
 
         # Getting start and end for query changes depending on command mode
-        start_time_epoch, end_time_epoch = self._get_time_period_borders(export_uuid,
-                                                                         analytic_object_metadata,
-                                                                         extraction_mode)
+        time_interval_dto = self._create_time_interval_dto(export_uuid, analytic_object_dto, extraction_mode)
+
+        self._setup_query_options(list_query_dto, extraction_mode, time_interval_dto)
 
         # Fetch analytic object changes
-        change_rows = self.change_fetcher.execute_query_list(query, start_time_epoch, end_time_epoch,
-                                                             self.record_modes[extraction_mode], filter_values)
+        protobuf_changes = self.change_fetcher.execute_query_list(list_query_dto)
 
         # Save changes to the database
-        self._save_to_db(analytic_object, filter_name, filter_values, query, properties, change_rows, extraction_mode)
+        self._save_to_db(list_query_dto, properties_dto, filter_name, filter_values, protobuf_changes, extraction_mode)
+
+    def _setup_query_options(self,
+                             list_query_dto: ListQueryExecutionDTO,
+                             extraction_mode: ExtractionMode,
+                             time_interval_dto: QueryTimeIntervalDTO) -> None:
+        list_query_dto.time_interval = time_interval_dto
+        if list_query_dto.options is None:
+            list_query_dto.options = ListQueryExecutionOptionsDTO()
+        list_query_dto.options.record_mode = self.record_modes[extraction_mode]
+        if list_query_dto.options.limit is None:
+            list_query_dto.options.limit = self.DEFAULT_QUERY_LIMIT
 
-    def _get_time_period_borders(self, export_uuid: str, analytic_object_metadata: dict[str, Any],
-                                 extraction_mode: ExtractionMode) -> Tuple[int, int]:
+    @staticmethod
+    def _create_list_query_dto(filter_values: Set[str], query: Dict[str, Any]) -> ListQueryExecutionDTO:
```

Comment:
How come the query is a Dict and not a (typed) DTO?

Reply:
This is because we don't have fully correct filters, and it will throw an exception when we try to create a DTO:
```
  "filters": [
    {
      "memberSet": {
        "dimension": {
          "name": "EmployeeID_Hierarchy",
          "qualifyingPath": "Employee"
        },
        "values": {
          "included": "${{EmployeeID}}"
        }
      }
    }
```

 I tried to use as little raw JSON (dict) as possible though.